cmd/service-controller/definition_monitor.go

```diff
@@ -829,6 +911,58 @@ func (m *DefinitionMonitor) processNextEvent() bool {
 						return fmt.Errorf("Failed to delete service definition on removal of previously annotated deployment %s: %s", name, err)
 					}
 				}
+			case "deploymentconfigs":
+				event.Recordf(DefinitionMonitorEvent, "deploymentconfig event for %s", name)
+				obj, exists, err := m.deploymentConfigInformer.GetStore().GetByKey(name)
+				if err != nil {
+					return fmt.Errorf("Error reading deploymentconfig %s from cache: %s", name, err)
+				} else if exists {
+					deploymentConfig, ok := obj.(*appv1.DeploymentConfig)
+					if !ok {
+						return fmt.Errorf("Expected Deployment for %s but got %#v", name, obj)
+					}
+
+					desired, ok := m.getServiceDefinitionFromAnnotatedDeploymentConfig(deploymentConfig)
+					if ok {
+						event.Recordf(DefinitionMonitorEvent, "Checking annotated deploymentconfig %s", name)
+						actual, ok := m.annotated[desired.Address]
+						if !ok || updateAnnotatedServiceDefinition(&actual, &desired) {
+							event.Recordf(DefinitionMonitorUpdateEvent, "Updating service definition for annotated deploymentconfig %s to %#v", name, desired)
+							changed := []types.ServiceInterface{
+								desired,
+							}
+							deleted := []string{}
+							err = kube.UpdateSkupperServices(changed, deleted, "annotation", m.vanClient.Namespace, m.vanClient.KubeClient)
+							if err != nil {
+								return fmt.Errorf("failed to update service definition for annotated deployment %s: %s", name, err)
+							}
+							m.annotated[desired.Address] = desired
+						}
+						address, ok := m.annotatedDeployments[name]
```

Comment:
This won't do the right thing if there is both a deployment and a deploymentconfig exposed and the have the same name. Though I suspect no one would want to do that deliberately, it could on a rare occasion happen by accident.

(Might be possible to collapse all the annotatedXXX maps into one, with the key including the type? Any new types would then be simpler to support)

Reply:
Updated in 4da22ec06fcfd70343546dab2a45b400b0c4c2a5